### PR TITLE
Feature/bake everything

### PR DESCRIPTION
**Pull request by steinkel**
Added a new "bake all everything" option.

The bake unit tests were in a funny state, so I'll push the additional unit test fixes once green is back in town...

### CODE REVIEW

File: src/Shell/BakeShell.php [hunk 1]
```diff
@@ -225,6 +225,7 @@ public function all($name = null)
         if (empty($name)) {
             $this->Model->connection = $this->connection;
             $this->out('Possible model names based on your database:');
+            $this->out('- everything');
```
Comment (ADmad) [inline src/Shell/BakeShell.php]: There should be some indication that this is a special name.

Comment (steinkel) [inline src/Shell/BakeShell.php]: Should I add it as a class constant?

Comment (steinkel) [inline src/Shell/BakeShell.php]: Could be an option instead too... like "--everything"

Comment (ADmad) [inline src/Shell/BakeShell.php]: An option would be better.

**Comment by markstory** [comment]
The tests [seem to be failing](https://travis-ci.org/cakephp/bake/jobs/64825354) because of your change.

File: src/Shell/BakeShell.php [hunk 2]
```diff
@@ -222,7 +222,7 @@ public function all($name = null)
             $this->connection = $this->params['connection'];
         }
 
-        if (empty($name)) {
+        if (empty($name) && !$this->params['everything']) {
```
Comment (markstory) [inline src/Shell/BakeShell.php]: If you use `param()` you can avoid the notice errors.

File: src/Shell/BakeShell.php [hunk 3]
```diff
@@ -232,15 +232,28 @@ public function all($name = null)
...
-            $this->{$task}->connection = $this->connection;
+        $allTables = collection([$name]);
+        $filteredTables = $allTables;
+
+        if ($this->params['everything']) {
+            $this->Model->connection = $this->connection;
+            $allTables = collection($this->Model->listAll());
+            $filteredTables = $allTables->reject(function ($tableName) {
+                return $tableName === 'phinxlog' || $tableName === 'users_phinxlog';
+            });
```
Comment (markstory) [inline src/Shell/BakeShell.php]: Do we want to toss out `i18n` and `cake_sessions` two?

File: src/Shell/BakeShell.php [hunk 4]
```diff
@@ -270,6 +283,10 @@ public function getOptionParser()
             ' are using command line arguments.'
         )->addSubcommand('all', [
             'help' => 'Bake a complete MVC skeleton.',
+        ])->addOption('everything', [
+            'help' => 'Bake a complete MVC skeleton, using all the available tables. ' .
+            'Usage: "bake all --everything true"',
```
Comment (markstory) [inline src/Shell/BakeShell.php]: You don't need to set true. You can set `'boolean' => true`.

**Comment by lorenzo** [comment]
Great job @steinkel !

**Pull request merged**